crates/runtime/src/http/v1/search.rs

```diff
@@ -56,7 +60,7 @@ pub(crate) async fn post(
     };
 
     match vs.search(&search_request).await {
-        Ok(resp) => match to_matches_sorted(&resp) {
+        Ok(resp) => match to_matches_sorted(&resp, search_request.limit) {
```

Comment:
Did we find that the number of results returned were > limit? This should be handled at the SQL level.

Reply:
The SQL limit is applied for single dataset/table search, when we have several datasets we should apply limit one more time after final grouping and sorting.